folium/utilities.py

```diff
@@ -345,3 +345,43 @@ def png_pack(png_tag, data):
         png_pack(b'IDAT', zlib.compress(raw_data, 9)),
         png_pack(b'IEND', b'')])
  
+def geodetic_to_mercator(geodetic):
+    """This function takes an 2D array in geodetic coordinates (ie: lat x
+    lon unprojected) and converts it to web mercator.  This is needed
+    to correctly overlay an image on a leaflet map, which uses web
+    mercator by default.
+
+    This code works with arrays that match the relative proportions of
+    latitude and longitude of the earth, meaning that they have twice
+    as many points in longitude as latitude (i.e.: (180, 360) for 1
+    degree resolution).
+
+    The code for this is from:
+    http://stackoverflow.com/questions/25058880/convert-to-web-mercator-with-numpy
+
+    Parameters
+    ----------
+    geodetic: 2D numpy array
+        Latitude x Longitude 2D array
+
+    meractor: 2d numpy array
+
+    """
+    
+    def row2lat(row):
+        return 180.0/np.pi*(2.0*np.atan(np.exp(row*np.pi/180.0))-np.pi/2.0)
```

Comment:
Bah, arctan. The original code used the math lib instead of bumpy.
On Wed, Aug 5, 2015 at 08:40 Filipe <notifications@github.com> wrote:

> In folium/utilities.py
> <https://github.com/python-visualization/folium/pull/168#discussion_r36296375>
> :
>
> > +    degree resolution).
> > +
> > +    The code for this is from:
> > +    http://stackoverflow.com/questions/25058880/convert-to-web-mercator-with-numpy
> > +
> > +    Parameters
> > +    ----------
> > +    geodetic: 2D numpy array
> > +        Latitude x Longitude 2D array
> > +
> > +    meractor: 2d numpy array
> > +
> > +    """
> > +
> > +    def row2lat(row):
> > +        return 180.0/np.pi*(2.0*np.atan(np.exp(row*np.pi/180.0))-np.pi/2.0)
>
> Do you mean np.arctan or np.arctan2?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/python-visualization/folium/pull/168/files#r36296375>.
>